packages/uikit-workshop/src/scripts/lit-components/pl-viewport-size-list/pl-viewport-size-list.js

```diff
@@ -74,20 +74,27 @@ class ViewportSizes extends BaseComponent {
           );
           break;
         case 'large':
+          // Do not evaluate a number higher than the clientWidth of the Iframe
+          // to prevent having max size multiple times
+          const max =
+            maxViewportWidth > this.iframe.clientWidth
+              ? this.iframe.clientWidth
+              : maxViewportWidth;
+
           this.iframe.fullMode = false;
           this.iframe.sizeiframe(
             getRandom(
               window.config.ishViewportRange !== undefined
                 ? parseInt(window.config.ishViewportRange.l[0], 10)
                 : 800,
-              maxViewportWidth
+              max
             ),
             true
           );
           break;
         case 'full':
           this.iframe.fullMode = true;
-          this.iframe.sizeiframe(maxViewportWidth, true);
+          this.iframe.sizeiframe(this.iframe.clientWidth, true);
```

Comment:
I'm a bit iffy on the changes in this file... could you elaborate a bit more on the change in behavior here?

Let's say the user sets their max viewport width config to `1200px` but is viewing PL on a `2560px` wide display... if they hit the `full` button, shouldn't Pattern Lab's viewport resizer width max out at `1200px` vs resizing to `2560px`?

@bmuenzenmeyer @bradfrost thoughts here?

Reply:
Shure, I can add some insights about my thoughts. I always felt that the `full` button is something like a reset to a state my screen supports. And that also means if I set large to a maximum of 1200px the full button will still scale to 2560px when I click on it. Also, there are cases where I need to test this behavior, especially when I want to test if my containers work correctly and even if the screen has 4k resolution, they do not size over a specific point.

Reply:
Little addition: I just found the file `view-sizes.js` there is a check if the value `window.config.ishMaximum` exists. I never saw this value in any `pl-config` (just available in `cli/test/util/patternlab-config.json`) file that's why I wasn't aware of it. Now it makes more sense and this changed line is invalid.

Reply:
I think I can help here.

The [Ish](https://bradfrost.com/blog/post/ish/) feature is not about setting a maximum that may never be exceeded. We all know how futile it is to imagine a world where we control the viewport of our users.

quoting from https://patternlab.io/docs/advanced-config-options.html

#### ishViewportRange
Sets the boundaries of each of the viewport toggles, `S`mall, `M`edium, and `L`arge. Clicking on one of these buttons will randomly set the ish Viewport to a value within the given range. Setting the range to the same number can effectively set an exact value. The first entry in `ishViewportRange.s` is the `ishViewportMinimum`, which is now obsolete. The second entry in `ishViewportRange.l` is the `ishViewportMaximum`, which is now also obsolete.

-----

to me, full has always meant full-screen, regardless of ish setting.